src/main/java/duke/CommandMatcher.java

```diff
@@ -0,0 +1,47 @@
+package duke;
+
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+/**
+ * This class serves as a way to abstract the idea of making a command
+ * as a matching process and an action.
+ */
+public class CommandMatcher {
+    private Predicate<String> shouldRunAction;
+    private Consumer<String> action;
+
+    /**
+     * Creates an object that handles checking and executing a command.
+     * @param shouldRunAction Predicate to check if the command should be run.
+     * @param action Action to run.
+     */

```

Comment:
For JavaDocs, there should be an empty line between description and parameter section. No issues otherwise!

Reply:
Thanks - made those changes!